tests/test_examples/test_io/test_ioresult_container/test_ioresult_pattern_matching.py

```diff
@@ -1,9 +1,13 @@
 from returns.io import IOFailure, IOResult, IOSuccess
+from returns.result import Success
 
 container: IOResult[int, str] = IOSuccess(42)
 match container:
     # Matches if the result stored inside `IOSuccess` is `42`
-    case IOSuccess(42):
+    # We need to use `Success` until the custom matching protocol
+    # is released. For more information, please visit:
+    # https://www.python.org/dev/peps/pep-0622/#custom-matching-protocol
+    case IOSuccess(Success(42)):
```

Comment:
@sobolevn I think we can make a workarround to make possible to do:
```python
match container:
    case IOSuccess(42):
        ...
```

After each `bind`, `map`, and etc we save the value in another variable than `_inner_value` and put it in `__match_args__` instead of `_inner_value`!
But also, we can wait for the __Custom Matching Protocol__!